pandas/core/generic.py

```diff
@@ -6469,15 +6469,45 @@ def fillna(
         else:
             return result.__finalize__(self, method="fillna")
 
+    @overload
+    def ffill(
+        self: FrameOrSeries,
+        axis: None | str | int = ...,
+        inplace: Literal[False] = ...,
+        limit: None | int = ...,
+        downcast: None | str | dict = ...,
+    ) -> FrameOrSeries:
+        ...
+
+    @overload
+    def ffill(
+        self: FrameOrSeries,
+        axis: None | str | int = ...,
+        inplace: Literal[True] = ...,
+        limit: None | int = ...,
+        downcast: None | str | dict = ...,
+    ) -> None:
+        ...
+
+    @overload
+    def ffill(
+        self: FrameOrSeries,
+        axis: None | str | int = ...,
+        inplace: bool_t = ...,
+        limit: None | int = ...,
+        downcast: None | str | dict = ...,
+    ) -> FrameOrSeries | None:
+        ...
+
     @final
     @doc(klass=_shared_doc_kwargs["klass"])
     def ffill(
         self: FrameOrSeries,
-        axis=None,
+        axis: None | str | int = None,
```

Comment:
axis has it's own type alias, `Axis` (check the fillna signature)

Reply:
Done